library/std/src/sync/rwlock.rs

```diff
@@ -25,7 +25,16 @@ use crate::sys_common::rwlock as sys;
 /// system's implementation, and this type does not guarantee that any
 /// particular policy will be used. In particular, a writer which is waiting to
 /// acquire the lock in `write` might or might not block concurrent calls to
-/// `read`.
+/// `read`, e.g.:
+///
+/// ```text
```

Comment:
I'm not convinced we spend a bunch of vertical space on a fairly obscure edge case, but I'd defer to @rust-lang/docs.

Reply:
I think it's up to you and libs; we don't have any real precedent here in either direction.

Reply:
Perhaps we can get the best of both worlds: is there a way to collapse a subset of the docs by default? It would be useful for detailed examples, e.g. like we do in GitHub:

<details>
<summary>Time-threads diagram</summary>

```rust
// Thread 1             |  // Thread 2
let _rg = lock.read();  |
                        |  // will block
                        |  let _wg = lock.write();
// may deadlock         |
let _rg = lock.read();  |
```
</details>

Reply:
This is with the `<details>` tag, which, given that markdown is a superset of HTML, should be usable. I don't think we currently use it at all, though.

Reply:
Tested it, and it works nicely, although it needs a small tweak: see #82805.